### ISSUE TITLE
ClassNotFoundException: net.bytebuddy.jar.asm.commons.ModuleHashesAttribute

### ISSUE THREAD

**Issue by suztomo** [raphw/byte-buddy#608]
I think Byte Buddy may be missing `net.bytebuddy.jar.asm.commons.ModuleHashesAttribute` in its jar. 

The class is referenced by ClassRemapper, which is included in the jar via [maven-shade-plugin configuration](https://github.com/raphw/byte-buddy/blob/b6822955a7d82d820bb449d89a42413efef9a283/byte-buddy/pom.xml#L114).

Code to produce `ClassNotFoundException`:

```
package net.bytebuddy.jar.asm;
// This package is to access Attribute class.

import net.bytebuddy.jar.asm.commons.ClassRemapper;

public class App {

  public static void main(String[] args) throws Exception {
    ClassRemapper classRemapper = new ClassRemapper(null, null);
    // This throws ClassNotFoundException: net.bytebuddy.jar.asm.commons.ModuleHashesAttribute
    classRemapper.visitAttribute(new Attribute("hello"));
  }
}
```

pom.xml

```
<?xml version="1.0" encoding="UTF-8"?>

<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
  <modelVersion>4.0.0</modelVersion>

  <groupId>com.google.suztomo</groupId>
  <artifactId>hello-bytebuddy</artifactId>
  <version>1.0-SNAPSHOT</version>

  <name>hello-bytebuddy</name>

  <properties>
    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
    <maven.compiler.source>1.8</maven.compiler.source>
    <maven.compiler.target>1.8</maven.compiler.target>
  </properties>

  <dependencies>
    <dependency>
      <groupId>net.bytebuddy</groupId>
      <artifactId>byte-buddy</artifactId>
      <version>1.9.2</version>
    </dependency>
  </dependencies>
</project>

```

![image](https://user-images.githubusercontent.com/28604/52233759-eb053e80-288d-11e9-849d-564b9c7e9960.png)

Note that I didn't encounter the error when only using Byte Buddy's public API (yet).

Do you think it's something to be fixed, or the class `ModuleHashesAttribute` intentionally omitted?

**Comment by raphw** [comment]
This is an oversight, I just did never test the shaded jar properly, sorry for that. I just fixed this and will make a new release some time soon.

**Comment by suztomo** [comment]
Thank you for confirmation.

**Comment by fwbrasil** [comment]
Should this issue be fixed on `1.9.9`? I getting the same error message:

```
[error] Exception in thread "Thread-2" java.lang.NoClassDefFoundError: net/bytebuddy/jar/asm/commons/ModuleHashesAttribute
[error] 	at net.bytebuddy.jar.asm.commons.ClassRemapper.visitAttribute(ClassRemapper.java:120)
[error] 	at net.bytebuddy.jar.asm.ClassReader.accept(ClassReader.java:642)
[error] 	at net.bytebuddy.jar.asm.ClassReader.accept(ClassReader.java:391)
[error] 	at net.bytebuddy.dynamic.scaffold.TypeWriter$Default$ForInlining.create(TypeWriter.java:3393)
[error] 	at net.bytebuddy.dynamic.scaffold.TypeWriter$Default.make(TypeWriter.java:1930)
[error] 	at net.bytebuddy.dynamic.scaffold.inline.RedefinitionDynamicTypeBuilder.make(RedefinitionDynamicTypeBuilder.java:217)
[error] 	at net.bytebuddy.dynamic.scaffold.inline.AbstractInliningDynamicTypeBuilder.make(AbstractInliningDynamicTypeBuilder.java:120)
[error] 	at net.bytebuddy.dynamic.DynamicType$Builder$AbstractBase.make(DynamicType.java:3396)
[error] 	at kyo.Kyo$$anon$1.reload$1(Kyo.scala:317)
[error] 	at kyo.Kyo$$anon$1.run(Kyo.scala:321)
[error] Caused by: java.lang.ClassNotFoundException: net.bytebuddy.jar.asm.commons.ModuleHashesAttribute
[error] 	at java.net.URLClassLoader.findClass(URLClassLoader.java:382)
[error] 	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
[error] 	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:349)
[error] 	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
[error] 	... 10 more
```

I'm using the library from Scala

```scala
        def reload[T, U](implicit t: ClassTag[T], u: ClassTag[U]) =
          (new ByteBuddy).redefine(t.runtimeClass).name(u.runtimeClass.getName).make().load(t.runtimeClass.getClassLoader, ClassReloadingStrategy.fromInstalledAgent())

        ByteBuddyAgent.install

        reload[UCFunction0, OCFunction0]
        reload[UCFunction1, OCFunction1]
        reload[UCFunction2, OCFunction2]
        reload[Kyo.type, Kyo.type]
```

**Comment by raphw** [comment]
No, i 1.9.9, I am using the milestone as a marker for when the issue was found to keep track, not really intuitive if you do not know it, sorry, I cannot change the labels.

I'll release 1.9.10 today.
